File: app/src/main/java/quickbeer/android/features/beerdetails/BeerReviewsView.java
/**
 * This file is part of QuickBeer.
 * Copyright (C) 2016 Antti Poikela <antti.poikela@iki.fi>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package quickbeer.android.features.beerdetails;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;
import android.widget.FrameLayout;

import java.util.List;

import quickbeer.android.R;
import quickbeer.android.data.pojos.Review;
import quickbeer.android.listeners.LoadMoreListener;

import static io.reark.reark.utils.Preconditions.checkNotNull;
import static io.reark.reark.utils.Preconditions.get;

public class BeerReviewsView extends FrameLayout {

    private BeerReviewsAdapter beerReviewsAdapter;

    private RecyclerView beerReviewsListView;

    public BeerReviewsView(Context context) {
        super(context);
    }

    public BeerReviewsView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void setReviews(@NonNull List<Review> reviews) {
        get(beerReviewsAdapter).setReviews(get(reviews));
    }

    public void setOnScrollListener(@NonNull LoadMoreListener listener) {
        checkNotNull(beerReviewsListView);
        checkNotNull(listener);

        listener.setLayoutManager((LinearLayoutManager) beerReviewsListView.getLayoutManager());
        beerReviewsListView.addOnScrollListener(listener);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();

        beerReviewsAdapter = new BeerReviewsAdapter();

        beerReviewsListView = (RecyclerView) findViewById(R.id.beers_reviews_list_view);
        beerReviewsListView.setHasFixedSize(true);
        beerReviewsListView.setLayoutManager(new LinearLayoutManager(getContext()));
        beerReviewsListView.setAdapter(beerReviewsAdapter);
    }
}

File: app/src/main/java/quickbeer/android/features/beerdetails/BeerReviewsView.kt
/**
 * This file is part of QuickBeer.
 * Copyright (C) 2017 Antti Poikela <antti.poikela@iki.fi>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package quickbeer.android.features.beerdetails

import android.content.Context
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.AttributeSet
import android.widget.FrameLayout
import quickbeer.android.R
import quickbeer.android.data.pojos.Review
import quickbeer.android.listeners.LoadMoreListener

class BeerReviewsView : FrameLayout {

    private lateinit var beerReviewsAdapter: BeerReviewsAdapter

    private lateinit var beerReviewsListView: RecyclerView

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    override fun onFinishInflate() {
        super.onFinishInflate()

        beerReviewsAdapter = BeerReviewsAdapter()

        beerReviewsListView = findViewById(R.id.beers_reviews_list_view) as RecyclerView
        beerReviewsListView.setHasFixedSize(true)
        beerReviewsListView.layoutManager = LinearLayoutManager(context)
        beerReviewsListView.adapter = beerReviewsAdapter
    }

    fun setReviews(reviews: List<Review>) {
        beerReviewsAdapter.setReviews(reviews)
    }

    fun setOnScrollListener(listener: LoadMoreListener) {
        listener.setLayoutManager(beerReviewsListView.layoutManager as LinearLayoutManager)
        beerReviewsListView.addOnScrollListener(listener)
    }
}
